src/csharp/Grpc.Core/Internal/PlatformInvocationWin64.cs

```diff
@@ -0,0 +1,771 @@
+
+        ChannelState IPlatformInvocation.grpcsharp_channel_check_connectivity_state(ChannelSafeHandle channel, int tryToConnect)
+        {
+            return grpcsharp_channel_check_connectivity_state(channel, tryToConnect);
+        }
+
+        void IPlatformInvocation.grpcsharp_channel_watch_connectivity_state(ChannelSafeHandle channel,
+            ChannelState lastObservedState, Timespec deadline, CompletionQueueSafeHandle cq, BatchContextSafeHandle ctx)
+        {
+            grpcsharp_channel_watch_connectivity_state(channel, lastObservedState, deadline, cq, ctx);
+        }
+
+        CStringSafeHandle IPlatformInvocation.grpcsharp_channel_get_target(ChannelSafeHandle call)
+        {
+            return grpcsharp_channel_get_target(call);
+        }
+
+        void IPlatformInvocation.grpcsharp_channel_destroy(IntPtr channel)
+        {
+            grpcsharp_channel_destroy(channel);
+        }
+
+
+        // CompletionQueueEvent
+
+        [DllImport("grpc_csharp_ext-x64.dll")]
+        static extern int grpcsharp_sizeof_grpc_event();
+
+        int IPlatformInvocation.grpcsharp_sizeof_grpc_event()
+        {
+            return grpcsharp_sizeof_grpc_event();
+        }
+
+
+        // CompletionQueueSafeHandle
+
+        [DllImport("grpc_csharp_ext-x64.dll")]
+        static extern CompletionQueueSafeHandle grpcsharp_completion_queue_create();
+
+        [DllImport("grpc_csharp_ext-x64.dll")]
+        static extern void grpcsharp_completion_queue_shutdown(CompletionQueueSafeHandle cq);
+
+        [DllImport("grpc_csharp_ext-x64.dll")]
+        static extern CompletionQueueEvent grpcsharp_completion_queue_next(CompletionQueueSafeHandle cq);
+
+        [DllImport("grpc_csharp_ext-x64.dll")]
+        static extern CompletionQueueEvent grpcsharp_completion_queue_pluck(CompletionQueueSafeHandle cq, IntPtr tag);
+
+        [DllImport("grpc_csharp_ext-x64.dll")]
+        static extern void grpcsharp_completion_queue_destroy(IntPtr cq);
+
+        CompletionQueueSafeHandle IPlatformInvocation.grpcsharp_completion_queue_create()
+        {
+            return grpcsharp_completion_queue_create();
+        }
+
+        void IPlatformInvocation.grpcsharp_completion_queue_shutdown(CompletionQueueSafeHandle cq)
+        {
+            grpcsharp_completion_queue_shutdown(cq);
+        }
+        CompletionQueueEvent IPlatformInvocation.grpcsharp_completion_queue_next(CompletionQueueSafeHandle cq)
+        {
+            return grpcsharp_completion_queue_next(cq);
+        }
+
+        CompletionQueueEvent IPlatformInvocation.grpcsharp_completion_queue_pluck(CompletionQueueSafeHandle cq,
+            IntPtr tag)
+        {
+            return grpcsharp_completion_queue_pluck(cq, tag);
+        }
+
+        void IPlatformInvocation.grpcsharp_completion_queue_destroy(IntPtr cq)
+        {
+            grpcsharp_completion_queue_destroy(cq);
+        }
+
+
+
+        // CStringSafeHandle 
+
+        [DllImport("grpc_csharp_ext-x64.dll")]
```

Comment:
Ah, I see what this trick was to differentiate between 32bit and x64.
This seems like one of the possible approaches, but I was actually thinking of doing something along the path of  inspecting what the size of   IntPtr is (4 or 8) and inspecting the current platform (Linux vs Mac vs Windows) and selecting the right shared library programmatically ("dlopen" or similar).
Once the shared library is loaded, all the PInvokes should work fine and .NET won't go ahead to search for a library that was already loaded.
What do you think?